ffi/src/lib.rs

```diff
@@ -193,20 +339,43 @@ pub unsafe extern "C" fn store_query<'a>(store: *mut Store, query: *const c_char
     Box::into_raw(Box::new(query_builder))
 }
 
+/// Binds a `TypedValue::Long` to a `Variable` with the given name.
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the `QueryBuilder` is not dangling and that
+/// the C string provided to `var` is valid.
 #[no_mangle]
 pub unsafe extern "C" fn query_builder_bind_long(query_builder: *mut QueryBuilder, var: *const c_char, value: c_longlong) {
     let var = c_char_to_string(var);
     let query_builder = &mut*query_builder;
    query_builder.bind_long(&var, value);
```

Comment:
Again, we allocate a `String` and then bind by reference.  I hope we can avoid a lot of this allocation with a little thought.